packages/react-composites/src/composites/CallComposite/components/SvgWithWordWrapping.tsx

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { _pxToRem } from '@internal/acs-ui-common';
+import React, { useLayoutEffect, useRef, useState } from 'react';
+
+/**
+ * An SVG element component that wraps inner text to fit the width of the SVG.
+ * @private
+ */
+export const SvgWithWordWrapping = (props: {
+  width: number;
+  text: string;
+  lineHeightPx: number;
+  bufferHeightPx: number;
+}): JSX.Element => {
+  const { width, text, lineHeightPx, bufferHeightPx } = props;
+  const svgRef = useRef<SVGSVGElement>(null);
+  const calculationTextElement = useRef<SVGTextElement>(null);
+  const visibleTextElement = useRef<SVGTextElement>(null);
+  const [height, setHeight] = useState<number>(0);
+
+  // useLayoutEffect ensures that the calculationTextElement is rendered before being used for calculations.
+  // Using useLayoutEffect over useEffect ensures we do not get a layout shift when the visibleTextElement is rendered
+  // and the height is updated. This is because useLayoutEffect runs synchronously after DOM mutations but
+  // before the browser has a chance to paint. See https://reactjs.org/docs/hooks-reference.html#uselayouteffect
+  // for more details.
+  useLayoutEffect(() => {
+    if (text && calculationTextElement.current && visibleTextElement.current) {
+      const numLines = convertTextToWrappedText(
+        calculationTextElement.current,
+        visibleTextElement.current,
+        width,
+        lineHeightPx
+      );
+      setHeight(numLines * lineHeightPx);
+    }
+  }, [width, lineHeightPx, text]);
+
+  return (
+    <svg width={width} height={height + bufferHeightPx} ref={svgRef} xmlns="http://www.w3.org/2000/svg">
+      <text height={0} ref={calculationTextElement} style={{ visibility: 'hidden' }}>
+        {text}
+      </text>
+      <text ref={visibleTextElement} x="0" y={bufferHeightPx / 2} role="heading" aria-level={1} />
+    </svg>
+  );
+};
+
+/**
+ * Wrap text in tspan elements to fit the width of the SVG
+ * @param baseTextElement The text element to create the wrapped text from.
+ * @param outputTextElement The text element to insert the wrapped text into.
+ * @param maxWidth The maximum width of the text element.
+ * @param lineHeightPx The height of each line in pixels.
+ * @returns The number of lines of text.
+ */
+const convertTextToWrappedText = (
+  inputTextElement: SVGTextElement,
+  outputTextElement: SVGTextElement,
+  maxWidth: number,
+  lineHeightPx: number
+): number => {
+  const words = inputTextElement.textContent?.split(' ') ?? [];
```

Comment:
Would a string like 'c  a t' be okay here?
Tested in playcode
```
'c  a t'.split(' ')
```
returns
```
["c", "", "a", "t"]
```
I'm worried it may unnecessarily add a tspan for the empty string

Reply:
Actually nvm this is fine. the empty string will add no width to the testLine. Disregard this.

Reply:
(Great question though, I hadn't tested this edge case)